### PR TITLE
demo_playback: Extend demo name length

### PR DESCRIPTION
**Pull request by khanghugo**

### CODE REVIEW

File: src/modules/demo_playback.rs [hunk 1]
```diff
@@ -204,8 +204,18 @@ pub fn set_next_demo(marker: MainThreadMarker) {
...
+
+                // Assigning each row with sliced string
+                while len > 16 {
+                    cls_demos.demos[count].copy_from_slice(&demo[(count * 16)..((count + 1) * 16)]);
+                    len = len - 16;
+                    count = count + 1;
+                }
+                cls_demos.demos[count][..len]
+                    .copy_from_slice(&demo[(count * 16)..(count * 16 + len)]);
+                cls_demos.demos[count + 1][0] = 0;
```
Comment (khanghugo) [inline src/modules/demo_playback.rs:218]: Not sure how necessary this line 218 is. I take it out of my build and it still works. This line could cause segfault.

Comment (khanghugo) [reply src/modules/demo_playback.rs:218]: Thing is I don't think anyone would make a demo name 512 characters long so maybe this is fine until someone decides to just try it out.

File: src/modules/demo_playback.rs [hunk 2]
```diff
@@ -204,8 +204,17 @@ pub fn set_next_demo(marker: MainThreadMarker) {
...
+                let mut len = demo.len();
+                let mut count = 0;
+
+                // Assigning each row with sliced string
+                while len > 16 {
+                    cls_demos.demos[count].copy_from_slice(&demo[(count * 16)..((count + 1) * 16)]);
+                    len = len - 16;
+                    count = count + 1;
+                }
+                cls_demos.demos[count][..len]
```
Comment (khanghugo) [inline src/modules/demo_playback.rs:216]: Maybe the second subscript can be culled also

**Comment by YaLTeR** [comment]
Rust has a lot of iterators and encourages their use; you'll frequently get better optimized code when you use the iterators. Check how I implemented this.

**Comment by khanghugo** [comment]
Yea, I was thinking of this but it won't look like that anyway. Thanks.

**Pull request merged**